src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -393,6 +393,63 @@ bool RegionNode::is_unreachable_from_root(const PhaseGVN* phase) const {
   return true; // The Region node is unreachable - it is dead.
 }
 
+bool RegionNode::is_self_loop(Node* n) {
```

Comment:
A bit expensive to DFS the entire graph to find a self loop.  You don't need to visit nodes outside the loop. But you might not need to do this at all - see my comments further down.

Reply:
DONE